raft/src/test/java/org/apache/kafka/raft/MockLogTest.java

```diff
@@ -750,4 +782,13 @@ private void appendBatch(int numRecords, int epoch) {
 
         appendAsLeader(records, epoch);
     }
+
+    private void appendBatch(int numRecords, int epoch, long initialOffset) {
```

Comment:
How about `private void appendBatch(int numberOfRecords, int epoch)` and always use the LEO like `appendAsLeader(Collection<SimpleRecord> records, int epoch)`?